src/options/JsonConfig.js

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import parserBabylon from "prettier/parser-babylon";
+import prettier from "prettier/standalone";
+import { validateOptions } from "./options";
+
+const SAVED_TIMEOUT = 500;
+const STRINGIFY_SPACING = 2;
+
+export default function JsonConfig({ errors, options, setErrors, setOptions }) {
+  const textareaEl = useRef(null);
+  const [textAreaVal, setTextAreaVal] = useState(
+    JSON.stringify(options),
+    null,
+    STRINGIFY_SPACING
+  );
+  const [displaySaved, setDisplaySaved] = useState(false);
+
+  useEffect(() => {
+    try {
+      setTextAreaVal(
+        prettier.format(JSON.stringify(options), {
+          parser: "json",
+          plugins: [parserBabylon],
+          ...options
+        })
+      );
+    } catch {}
```

Comment:
If this throws when the text field is invalid, I'd like to show (or at least log) an error.

Reply:
What would you want the error to say? I’m not sure how useful an error here would actually be to the user. That being said, I understand the fear of swallowing errors silently. I guess we could log something in the console?

Reply:
Could be a simple "Error: Invalid JSON at ..." message in the form

Reply:
That should be covered already though, since we JSON parse it. This was more for safety than anything else - didn't want an internal error in Prettier to bubble up and break the rest of the extension.

Reply:
Can we table this for now and see if it's really an issue? Like I said, this was more so that some error out of our control couldn't bring the whole extension down.

Reply:
I'm not opposed to adding something here, but I do want to get this merged so we can continue towards v1!

Reply:
Yea that's fine with me, I resolved this. I think I misread the code at first, it's not that unsafe.